java/algorithms/MergeSortIterative-FunSat02/Main.java

```diff
@@ -0,0 +1,134 @@
+import org.sosy_lab.sv_benchmarks.Verifier;
+
+/**
+ * Type             : Functional Safety
+ * Expected Verdict : True
+ * Last modified by : Zafer Esen <zafer.esen@it.uu.se>
+ * Date             : 11 November 2019
+ *
+ * This benchmark is just like IterativeMergeSort-FunSat01, but
+ * with a stronger sortedness assertion.
+ *
+ * Permission to freely use and modify the file granted via e-mail on 26.10.2019
+ * by the original author David Kosbie <koz@cmu.edu>.
+ */
+
+// IterativeMergeSort.java
+// By David Kosbie
+
+// To see iterative mergesort in action, check out the (really great!) xSortLab:
+//   http://math.hws.edu/TMCM/java/xSortLab/
+// Select "mergesort" and step through the algorithm visually.  Neat!
+
+// How it works:  on the first pass, we merge a[0] and a[1], then we merge
+// a[2] and a[3], and so on.  So the array of n elements contains n/2 sorted
+// subarrays of size 2.  On the second pass, we merge a[0]-a[3], a[4]-a[7],
+// and so on, so the array of n elements contains n/4 sorted subarrays of size
+// 4.  We keep doubling our "blockSize" until it reaches n, and we're done.
+
+public class Main {
+
+  public static void main(String[] args) {
+    final int N = Verifier.nondetInt();
+
+    if (N < 1) return;
```

Comment:
nitpick: We might use Verifier.assume here as well.

Reply:
Done, updated this in the rest of the MergeSortIterative benchmarks as well.